File: app/src/main/java/com/skyd/imomoe/model/interfaces/IRankModel.java
package com.skyd.imomoe.model.interfaces;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.skyd.imomoe.bean.TabBean;

import java.util.ArrayList;

/**
 * 获取排行榜界面每个Tab详细数据的接口
 */
public interface IRankModel extends IBase {
    String implName = "RankModel";

    /**
     * 获取排行榜Tab数据
     *
     * @param callBack 回调接口，不为null。若同步获取数据，请直接return返回，忽略callBack；
     *                 若异步获取数据，则return null，使用callBack返回数据
     * @return ArrayList，可为null。排行榜TabArrayList
     */
    @Nullable
    ArrayList<TabBean> getRankTabData(@NonNull RankTabDataCallBack callBack);

    interface RankTabDataCallBack {
        void onSuccess(@NonNull ArrayList<TabBean> list);

        void onError(@NonNull Exception e);
    }
}

File: plugin/src/main/java/com/su/mediabox/plugin/interfaces/IRankModel.kt
package com.su.mediabox.plugin.interfaces

import com.su.mediabox.plugin.standard.been.TabBean

/**
 * 获取排行榜界面每个Tab详细数据的接口
 */
interface IRankModel : IBase {
    /**
     * 获取排行榜Tab数据
     *
     * @return ArrayList，不可为null。排行榜TabArrayList
     */
    suspend fun getRankTabData(): ArrayList<TabBean>

    companion object {
        const val implName = "RankModel"
    }
}